### PR TITLE
Indilib: add support for third party drivers

### PR DESCRIPTION
**Pull request by hjones2199**
###### Motivation for this change
Indilib has moved many of its astronomy equipment drivers to a seperate repository (indi-3rdparty). To support these drivers (and arbitrary addition of other indi drivers) in nixpkgs, a mechanism to wrap drivers into a single nix store path is required. This allows other nix derivations to point to a nix store path that has indi drivers from different sources (and derivations). Most importantly, kstars and phd2 can now utilize all astronomy equipment supported by the indi project.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 114586` at 18040bda run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>5 packages built:</summary>
  <ul>
    <li><a href="https://gist.github.com/3431ba0dce213a5f2df4714d7a0fd208">indi-3rdparty</a></li></li>
    <li><a href="https://gist.github.com/205f953a5e76c07fd29093d923f1b0a2">indi-full</a></li></li>
    <li><a href="https://gist.github.com/ed8b0cfe667c871f5dabf0f5f3002257">indilib</a></li></li>
    <li><a href="https://gist.github.com/afddbe942ee1139932ac3209e58d4009">kstars</a></li></li>
    <li><a href="https://gist.github.com/b351d76898fef9c56afa977dc130cabd">phd2</a></li></li>
  </ul>
</details>
<details>
  <summary>1 suggestion:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/unclear-gpl.md">unclear-gpl</a>

`gpl2` is a deprecated license, check if project uses `gpl2Plus` or `gpl2Only` and change `meta.license` accordingly.

Near pkgs/applications/science/astronomy/kstars/default.nix:56:5:
```
   |
56 |     license = licenses.gpl2;
   |     ^
```
</li>
  </ul>
</details>

File: pkgs/development/libraries/indilib/default.nix [hunk 1]
```diff
@@ -23,10 +24,6 @@ stdenv.mkDerivation rec {
     sha256 = "sha256-WTRfV6f764tDGKnQVd1jeYN/qXa/VRTFK0mMalc+9aU=";
   };
 
-  patches = [
-    ./udev-dir.patch
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/indilib/default.nix:27]: Please remove the file.

**Comment by SuperSandro2000** [comment]
This is a semi-automatic executed nixpkgs-review with [nixpkgs-review-checks extension](https://github.com/SuperSandro2000/nixpkgs-review-checks). It is checked by a human on a best effort basis and does not build all packages (e.g. lumo, tensorflow or pytorch).
If you have any questions or problems please reach out to SuperSandro2000 on IRC.

Result of `nixpkgs-review pr 114586` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
<summary>
5 packages built:
</summary>
<ul>
<li>indi-3rdparty</li>
<li>indi-full</li>
<li>indilib</li>
<li>kstars</li>
<li>phd2</li>
</ul>
</details>

The following issues got detected with the above build packages.
Please fix at least the ones listed with your changed packages:

<details>
indi-full:
<details>


warning: unused-argument
Unused argument: `paths`.
Near pkgs/build-support/trivial-builders.nix:284:12:

```
    |
284 |          , paths
    |            ^
```
</details>
</details>

**Review by TimPut — APPROVED** [review]
Nice work, works for me. Thanks for your work making nixpkgs better for astro!

**Pull request merged**